### ISSUE TITLE
bug: <video> element playback issues on iOS using capacitor:// protocol

### ISSUE THREAD

**Issue by chrisbrown-io** [ionic-team/capacitor#6790]
### Capacitor Version

```
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 5.2.2
  @capacitor/core: 5.2.2
  @capacitor/android: 5.2.2
  @capacitor/ios: 5.2.2

Installed Dependencies:

  @capacitor/android: not installed
  @capacitor/cli: 5.2.2
  @capacitor/core: 5.2.2
  @capacitor/ios: 5.2.2

[success] iOS looking great! 👌
```

### Platform(s)

iOS 16.5.1 (iPhone 12 mini)

### Expected Behavior

MP4 source in html `<video>` tag should play back when referencing local assets (internally using `capacitor://` protocol).

In practice, videos only seem to play back correctly when they're using http/https protocol, which forces using remote assets.

After some investiagion, I think this may be a Safari/iOS issue with how the `<video>` element makes the request under the hood. When requesting a video using a http/https source, iOS sends extra request headers that are not sent when using local assets that use the capacitor protocol. When using the capacitor protocol (for requesting local files), after the initial `Byte-Range 0-1` request, a number of seemingly random requests for 8 byte ranges are sent and the video never properly plays back. This is strange, as the response and `Content-range` header returns the correct range, and same value(s) for the exact same file.

I think this is either a Safari quirk or an issue (that I can't get to the bottom of) with how the `WebViewAssetHandler` is handling these requests. Although after digging through, the `WebViewAssetHandler` seems to be doing everything correctly.

Using the same [video asset ](https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4) and the same markup, you can see the difference depending on how the `src` attribute is specified in the `<video>`

Using HTTP:

![Screenshot 2023-08-04 at 15 07 26](https://github.com/ionic-team/capacitor/assets/1853199/7542bb29-3ea5-40ce-b5d1-3742d18a753f)

![Screenshot 2023-08-04 at 15 07 33](https://github.com/ionic-team/capacitor/assets/1853199/a1f01466-8c4b-4960-a936-5dac042e2c31)

Using `capacitor`:

![Screenshot 2023-08-04 at 14 44 59](https://github.com/ionic-team/capacitor/assets/1853199/bccded1f-c4fd-477f-9308-1ba8daf759bb)

![Screenshot 2023-08-04 at 14 43 06](https://github.com/ionic-team/capacitor/assets/1853199/694bc806-1201-4b86-ac36-1e9cde5d71fd)

### Code Reproduction

A basic reproduction can be found here:

https://github.com/chrisbrown-io/capacitor-ios-video-issue

If you clone the repo and install the deps and then run `cap sync` / `npm run sync` you can see the issue if you run on a device. The video asset in both `<video>` tags are using the same source, one hosted locally and one remotely.

To see how the issue is fixed by using http, you can host the local files with `sirv-cli` by running `npm run dev` and then update your `capacitor.config.js` file with the `server.url` option to point at the sirv http server, e.g:

```
module.exports = {
  ...
  "server": {
    ...
    "url": "http://192.168.1.1:8080"
  }
}
```

Then run `cap sync` / `npm run sync` and run another build on a device. Both `<video>` tags will then function as expected, as both are being served over http.

### Other Technical Details

`npm --version` output: `9.6.7`

`node --version` output: `v18.17.0`

`pod --version` output (iOS issues only): `1.12.0`

### Additional Context

I think this _could_ be related in some way to #6021 and the way the video requests are handled.

**Comment by chrisbrown-io** [comment]
On further investigation, this is definitely the same issue as #6721 .

After considerable further testing, my hypothesis is that this is a bug within iOS `16.5.x`.

I have tested this on a simulator running `16.4` and I have tested on a real device running `16.6` and the video loads/plays correctly and makes the correct HTTP requests matching the `http://`/`https://` source.

What would be the recommended approach for handling such a version specific OS bug as this? Detect version using capacitor APIs and implement alternative/different functionality?

**Comment by markemer** [comment]
> On further investigation, this is definitely the same issue as #6721 - apologies, I had not seen this ticket beforehand.
> 
> After more testing, my hypothesis is that this is a bug within iOS `16.5.x` (I'm unsure of which exact minor/patch version(s) this issue exists on).
> 
> I have tested this on a simulator running `16.4` and I have tested on a real device running `16.6` and the video loads/plays correctly and makes the correct HTTP requests matching the `http://`/`https://` source.
> 
> What would be the recommended approach for handling such a version specific OS bug as this? Detect version using capacitor APIs and implement alternative/different functionality?

So, this just fails on 16.5.x? That's strange. I'm not sure how we'd want to handle something that specific.

**Comment by TheRealPingWin** [comment]
Having the same issue, however also on ipad pro 5th gen, ios 15.3.1.
However for me first time it loads the video fully and after changing the src, it won't load the whole video only some bytes.
Changing the src again and again results in sometimes loading the whole video, sometimes just one frame.

**Comment by jcesarmobile** [comment]
I've been searching in webkit's issue tracker and webkit's github repository and I couldn't find any issue or commit that could be related, but their commits are a bit cryptic, so it's hard to find anything there and also they don't provide information about which commits land in which iOS versions.

But I agree that this is an iOS 16.5.x issue and it's fixed in 16.6, and there is nothing we can do about it.
Apple users usually update very fast, so I think you shouldn't worry about it. But if you are, then you can probably use the device plugin to get the iOS version and if they are in 16.5, tell them to update if they are having problems with videos.

The other issue about changing the src seems a different problem, please, create a new issue about it and provide a sample app that reproduces the issue, but also test in other iOS versions as it could also be specific to that version.
